impress-org/givewp
# Feature: implement necessary tweaks to make the Webhooks addon properly work

## Description



This PR implements the necessary tweaks to make some features implemented on the new GiveWP Webhooks addon work as expected. The most important change on PR is the Action Scheduler package added as a dependency from Give core, this is a vital package to make the new Webhooks addon work.

This PR also changes how the profile update made using the Donor Dashboard works. Instead of updating info (first name, last name, and email) once per time, we use models to update all at once. This way we can prevent that multiple Webhooks Events get triggered because of just one update.

To make the `subscription_updated` and `subscription_status_changed` events work through the legacy code, was necessary to add to the codebase the following hooks:

```
give_recurring_pre_update_subscription
give_recurring_update_subscription_status
```

## Affects



The new GiveWP Webhooks addon.

## Testing Instructions



1. Pull the changes from this PR
2. Pull the changes from the Webhooks addon, most precisely this branch: https://github.com/impress-org/give-webhooks/pull/7
3. Run the unit tests and make sure everything works
4. To test the donor dashboard update, just create a new webhook with the `donor_updated` event, update the first name, last name, and email using the donor dashboard, then go to `WP > Tools > Scheduled Actions` page and make sure that just one `donor_updated` event was created

## Pre-review Checklist
